src/PaymentMethod.js

```diff
@@ -128,16 +135,52 @@ export default class PaymentMethod extends Component {
     if (!method) {
       throw new Error('[PaymentMethod.props.defaultMethod missing from `PaymentMethod.props.methods`]')
     }
-
   }
 
   componentDidMount() {
-    const { account } = this.props
+    const {
+      account,
+      instrument = {},
+      edit = false,
+      getToken,
+    } = this.props
+
     this.setAccount(account)
+
+    // If editing an existing instrument, get auth then fetch instrument
+    if (edit === true && !!getToken !== false && typeof (getToken) === 'function') {
+      if(!!instrument.id === false){
+        console.warn("You must provide an instrument id to edit that instrument")
+        return
+      }
+      getToken(account.accountId)
+        .then(accessToken => {
+          InstrumentModel.fetchInstrument(account.id, instrument.id, accessToken)
+            .then(instrument => this.setupInstrument(instrument))
+        })
+        .catch(error => console.error("There was an error retrieving your instrument" + error))
```

Comment:
Using `console.error` and `console.warn` is generally frowned upon in library code. I know we haven't setup a style guide yet, but we should add a config option to show/hide logging and wrap these calls with a logger utility that pays attention to the context.